lib/assessment.rb

```diff
@@ -13,36 +13,47 @@ def longest(enum)
   # enumerable supplied (i.e. apply the `+` operator)
   # Assume each entry implements a `+` method
   def sum(enum)
-    # Your code goes here
+    enum.reduce(&:+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the numbers enumerable are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.any? { |x| x.odd? }
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # numbers enumerable are even
   def every_even(numbers)
-    # Your code goes here
+    numbers.all? { |x| x.even? }
   end
 
   # Question 5
   # Write ruby code that returns an array with all of the
   # entries from the words enumerable capitalized.
   def transform(words)
-    # Your code goes here
+    words.map(&:capitalize)
   end
 
+  # the reason why we can use the &:capitalize after words.map is
+  # because it operates the same was as words.map { |w| w.capitalize }
+
   # Question 6
   # Write ruby code to read each line from the file
   # named filename. Concatenate the lines together after
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    File.open(filename).reduce(' ') { |a, e| a << e.chomp + ' ' }.strip
```

Comment:
We probably want to start with an empty string here, not a space. But otherwise, this is correct.